src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -1264,29 +1272,71 @@ CATCH_RETURN();
             fallback = _fontRenderData->SystemFontFallback();
         }
 
-        // Walk through and analyze the entire string
-        while (textLength > 0)
+        ::Microsoft::WRL::ComPtr<IDWriteFontFallback1> fallback1;
+        ::Microsoft::WRL::ComPtr<IDWriteTextFormat3> format3;
+
+        // If the OS supports IDWriteFontFallback1 and IDWriteTextFormat3, we can use the
+        // newer MapCharacters to apply axes of variation to the font
+        if (!FAILED(_formatInUse->QueryInterface(IID_PPV_ARGS(&format3))) && !FAILED(fallback->QueryInterface(IID_PPV_ARGS(&fallback1))))
+        {
+            const auto axesVector = _fontRenderData->GetAxisVector(weight, stretch, style, format1->GetFontSize(), format3.Get());

```

Comment:
this is to support doing fallback while preserving the user's requested axes, right?

Reply:
Yes, if the user's requested axes were in the 'old' format (like `Condensed` for font stretch for example), `GetAxisVector` will convert it into the 'new' format (a number) for the new `MapCharacters` that gets called below